Index: Makefile.in
==================================================================
--- Makefile.in
+++ Makefile.in
@@ -194,11 +194,11 @@
          vdbetrace.lo vdbevtab.lo \
          wal.lo walker.lo where.lo wherecode.lo whereexpr.lo \
          window.lo utf.lo vtab.lo
 
 LIBOBJS0 += hct_pman.lo hct_record.lo hctree.lo hct_stats.lo
-LIBOBJS0 += hct_tree.lo hct_file.lo hct_database.lo hct_tmap.lo 
+LIBOBJS0 += hct_tree.lo hct_file.lo hct_database.lo hct_tmap.lo
 LIBOBJS0 += btwrapper.lo
 
 # Object files for the amalgamation.
 #
 LIBOBJS1 = sqlite3.lo
@@ -382,11 +382,15 @@
   $(TOP)/src/hctPManInt.h  \
   $(TOP)/src/hctTMapInt.h  \
   $(TOP)/src/btreeModules.h \
   $(TOP)/src/btreeDefine.h \
   $(TOP)/src/btreeUndef.h \
-  $(TOP)/src/btwrapper.c
+  $(TOP)/src/btwrapper.c \
+  $(TOP)/src/sqlite3hct.h \
+  $(TOP)/src/hctJrnlInt.h \
+  $(TOP)/src/hct_journal.c \
+  $(TOP)/src/hct_journalhash.c
 
 # Source code to the test files.
 #
 TESTSRC = \
   $(TOP)/src/test1.c \

Index: tool/mksqlite3c.tcl
==================================================================
--- tool/mksqlite3c.tcl
+++ tool/mksqlite3c.tcl
@@ -140,11 +140,11 @@
 }
 
 # Examine the parse.c file.  If it contains lines of the form:
 #
 #    "#ifndef SQLITE_ENABLE_UPDATE_LIMIT
-# 
+#
 # then set the SQLITE_UDL_CAPABLE_PARSER flag in the amalgamation.
 #
 set in [open $srcdir/parse.c]
 if {[regexp {ifndef SQLITE_ENABLE_UPDATE_DELETE_LIMIT} [read $in]]} {
   puts $out "#define SQLITE_UDL_CAPABLE_PARSER 1"
@@ -196,10 +196,12 @@
    hctTMapInt.h
    btreeModules.h
    btreeDefine.h
    btreeUndef.h
    sqlite3recover.h
+   sqlite3hct.h
+   hctJrnlInt.h
 } {
   set available_hdr($hdr) 1
 }
 set available_hdr(sqliteInt.h) 0
 set available_hdr(os_common.h) 0
@@ -486,11 +488,13 @@
    hct_file.c
    hct_database.c
    hct_tmap.c
    hct_record.c
    hct_stats.c
-} 
+   hct_journal.c
+   hct_journalhash.c
+}
 if {$enable_recover} {
   lappend flist sqlite3recover.c dbdata.c
 }
 foreach file $flist {
   copy_file $srcdir/$file

